include/oneapi/dpl/internal/random_impl/lognormal_distribution.h

```diff
@@ -0,0 +1,219 @@
+// -*- C++ -*-
+//===-- lognormal_distribution.h ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Lognormal Distribution
+
+#ifndef _ONEDPL_LOGNORMAL_DISTRIBUTION
+#define _ONEDPL_LOGNORMAL_DISTRIBUTION
+
+namespace oneapi
+{
+namespace dpl
+{
+template <class _RealType = double>
+class lognormal_distribution
+{
+  public:
+    // Distribution types
+    using result_type = _RealType;
+    using scalar_type = internal::element_type_t<_RealType>;
+
+    struct param_type
+    {
+        param_type() : param_type(scalar_type{0.0}) {}
+        param_type(scalar_type __mean, scalar_type __stddev = scalar_type{1.0}) : mean(__mean), stddev(__stddev) {}
+        scalar_type mean;

```

Comment:
should it also be named m and s as methods of the distribution, so the user won't be confused with different naming in case he want to do something like this: auto var1 = pt.mean; auto var2 = distr.m where pt is and object of param_type and distr is and object of lognormal_distribution type?

Reply:
done